matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java

```diff
@@ -0,0 +1,61 @@
+package org.matsim.core.router;
+
+import org.apache.log4j.Logger;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.api.core.v01.network.Network;
+import org.matsim.core.gbl.Gbl;
+import org.matsim.core.network.NetworkUtils;
+import org.matsim.facilities.FacilitiesUtils;
+import org.matsim.facilities.Facility;
+
+class MultimodalLinkChooserDefaultImpl implements MultimodalLinkChooser {
+
+    private static final Logger log = Logger.getLogger( FacilitiesUtils.class ) ;
+
+    @Override
+    public Link decideOnLink(Facility facility, Network network) {
+        Link accessActLink = null ;
+
+        Id<Link> accessActLinkId = null ;
+        try {
+            accessActLinkId = facility.getLinkId() ;
+        } catch ( Exception ee ) {
+            // there are implementations that throw an exception here although "null" is, in fact, an interpretable value. kai, oct'18
+        }
+
+        if ( accessActLinkId!=null ) {
+            accessActLink = network.getLinks().get( facility.getLinkId() );
+            // i.e. if street address is in mode-specific subnetwork, I just use that, and do not search for another (possibly closer)
+            // other link.
+
+        }
+
+        if ( accessActLink==null ) {
+            // this is the case where the postal address link is NOT in the subnetwork, i.e. does NOT serve the desired mode,
+            // OR the facility does not have a street address link in the first place.
+
+            if( facility.getCoord()==null ) {
+                throw new RuntimeException("link for facility cannot be determined when neither facility link id nor facility coordinate given") ;
+            }
+
+            accessActLink = NetworkUtils.getNearestLink(network, facility.getCoord()) ;
+            if ( accessActLink == null ) {
+                int ii = 0 ;
```

Comment:
What is this for loop doing? It looks like it is taking 10 links from the network and printing an error message for them?

Reply:
This is code copied over here from FacilitiesUtils (where we set the original code to deprecated). I have to admit that we just copied and now I'm wondering too whether this is just some debug output committed by accident into FacilitiesUtils.

Reply:
I talked to @kainagel about these lines of code. They have a debugging purpose. I added a comment in #2051 to explain.

Reply:
Ah, this makes it clearer. Copy paste development is the best kind of development 🙃